### PR TITLE
fix: refer to global schema definitions when creating models

### PR DESCRIPTION
**Pull request by alaeddine-13**
closes: https://github.com/jina-ai/jina/issues/6019

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]
Can we have a test for the specific issue, maube create a new file under docarray_v2 with test_issues.py where we add tests for specific issues.

File: tests/unit/serve/runtimes/test_helper.py [hunk 1]
```diff
@@ -108,26 +116,43 @@ class CustomDoc(BaseDoc):
         u: Union[str, int]
         lu: List[Union[str, int]] = [0, 1, 2]
         tags: Optional[Dict[str, Any]] = None
+        nested: NestedDoc
```
[the remark this answers is not in the document]
Comment (alaeddine-13) [reply tests/unit/serve/runtimes/test_helper.py]: @JoanFM  my bad the issue appears only at 2 or more levels of nesting. I updated the tests

File: tests/integration/docarray_v2/test_issues.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from typing import List, Optional
+
+from docarray import BaseDoc, DocList
+
+from jina import Executor, Flow, requests
+
+
+class Nested2Doc(BaseDoc):
```
Comment (JoanFM) [inline tests/integration/docarray_v2/test_issues.py:8]: can we copy the exact same structure as the issue?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/integration/docarray_v2/test_issues.py (hunk 2 above)
Comment (alaeddine-13) [reply tests/integration/docarray_v2/test_issues.py:8]: I think the only difference is the use of `Optional` and `List`. I added another case for this

**Pull request merged**